python/tests/test_adjoint_jax.py

```diff
@@ -22,228 +22,251 @@
-  gaussian_rel_width=0.2,
-  sim_resolution=20,
-  design_region_resolution=20,
+    wg_width=0.5,
+    wg_padding=1.0,
+    wg_length=1.0,
+    pml_width=1.0,
+    source_to_pml=0.5,
+    source_to_monitor=0.1,
+    frequencies=[1 / 1.55],
+    gaussian_rel_width=0.2,
+    sim_resolution=20,
+    design_region_resolution=20,
 ):
-  """Builds a simulation of a straight waveguide with a design region segment."""
-  design_region_shape = (1.0, wg_width)
-
-  # Simulation domain size
-  sx = 2 * pml_width + 2 * wg_length + design_region_shape[0]
-  sy = 2 * pml_width + 2 * wg_padding + max(
-    wg_width,
-    design_region_shape[1],
-  )
-
-  # Mean / center frequency
-  fmean = onp.mean(frequencies)
-
-  si = mp.Medium(index=3.4)
-  sio2 = mp.Medium(index=1.44)
-
-  sources = [
-    mp.EigenModeSource(
-      mp.GaussianSource(frequency=fmean, fwidth=fmean * gaussian_rel_width),
-      eig_band=1,
-      direction=mp.NO_DIRECTION,
-      eig_kpoint=mp.Vector3(1, 0, 0),
-      size=mp.Vector3(0, wg_width + 2 * wg_padding, 0),
-      center=[-sx / 2 + pml_width + source_to_pml, 0, 0],
+    """Builds a simulation of a straight waveguide with a design region segment."""
+    design_region_shape = (1.0, wg_width)
+
+    # Simulation domain size
+    sx = 2 * pml_width + 2 * wg_length + design_region_shape[0]
+    sy = 2 * pml_width + 2 * wg_padding + max(
+        wg_width,
+        design_region_shape[1],
     )
-  ]
-
-  nx = int(design_region_resolution * design_region_shape[0])
-  ny = int(design_region_resolution * design_region_shape[1])
-  mat_grid = mp.MaterialGrid(
-    mp.Vector3(nx, ny),
-    sio2,
-    si,
-    grid_type='U_DEFAULT',
-  )
-
-  design_regions = [
-    mpa.DesignRegion(
-      mat_grid,
-      volume=mp.Volume(
-        center=mp.Vector3(),
-        size=mp.Vector3(
-          design_region_shape[0],
-          design_region_shape[1],
-          0,
+
+    # Mean / center frequency
+    fmean = onp.mean(frequencies)
+
+    si = mp.Medium(index=3.4)
+    sio2 = mp.Medium(index=1.44)
+
+    sources = [
+        mp.EigenModeSource(
+            mp.GaussianSource(frequency=fmean,
+                              fwidth=fmean * gaussian_rel_width),
+            eig_band=1,
+            direction=mp.NO_DIRECTION,
+            eig_kpoint=mp.Vector3(1, 0, 0),
+            size=mp.Vector3(0, wg_width + 2 * wg_padding, 0),
+            center=[-sx / 2 + pml_width + source_to_pml, 0, 0],
+        ),
+        mp.EigenModeSource(
+            mp.GaussianSource(frequency=fmean,
+                              fwidth=fmean * gaussian_rel_width),
+            eig_band=1,
+            direction=mp.NO_DIRECTION,
+            eig_kpoint=mp.Vector3(-1, 0, 0),
+            size=mp.Vector3(0, wg_width + 2 * wg_padding, 0),
+            center=[sx / 2 - pml_width - source_to_pml, 0, 0],
         ),
-      ),
+    ]
+
+    nx = int(design_region_resolution * design_region_shape[0])
+    ny = int(design_region_resolution * design_region_shape[1])
```

Comment:
The `nx` and `ny` dimensions of the `MaterialGrid` are missing a `+ 1` (see #1760).

Reply:
Fixed.